app/src/main/java/ch/epfl/skysync/viewmodel/FlightsOverviewViewModel.kt

```diff
@@ -0,0 +1,73 @@
+package ch.epfl.skysync.viewmodel
+
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.compose.viewModel
+import ch.epfl.skysync.Repository
+import ch.epfl.skysync.models.UNSET_ID
+import ch.epfl.skysync.models.flight.Flight
+import ch.epfl.skysync.models.user.Admin
+import ch.epfl.skysync.models.user.Crew
+import ch.epfl.skysync.models.user.Pilot
+import ch.epfl.skysync.models.user.User
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+/** ViewModel for the user */
+class FlightsOverviewViewModel(
+    val repository: Repository,
+    val userId: String?,
```

Comment:
Here you have the same problem as in #258, this should be `userid: String` or to be consistent with the navigation graph: `uid: String`